cloud/cloudstack/cs_instance.py

```diff
@@ -547,6 +562,20 @@ def get_user_data(self):
             user_data = base64.b64encode(user_data)
         return user_data
 
+    def get_details(self):
+        res = None
+        cpu = self.module.params.get('cpu')
+        cpu_speed = self.module.params.get('cpu_speed')
+        memory = self.module.params.get('memory')
+        if all([cpu, cpu_speed, memory]):
+            res = [{
+                'cpuNumber': cpu,
+                'cpuSpeed': cpu_speed,
+                'memory': memory,
+            }]
+        elif any([cpu, cpu_speed, memory]):
```

Comment:
there is a ansible module feature, where you can specify which arguments must used together, see https://github.com/DazWorrall/ansible-modules-extras/blob/feature/cs_instance_custom/cloud/cloudstack/cs_instance.py#L858. 

~~~
required_together = (
     ['api_key', 'api_secret', 'api_url'],
     ['cpu', 'cpu_speed', 'memory'],
),
~~~